torch/nn/modules/transformer.py

```diff
@@ -0,0 +1,290 @@
+import torch
+import copy
+from .. import functional as F
+from .module import Module
+from .activation import MultiheadAttention
+from .container import ModuleList
+import numpy as np
+from ..init import xavier_uniform_
+from .dropout import Dropout
+from .linear import Linear
+from .normalization import LayerNorm
+
+class TransformerBase(Module):
```

Comment:
do you really need a base when there is only one subclass?

Reply:
This base class just lays out the encoder-decoder architecture.

Reply:
Let's remove this until we need an abstraction. Just like @SsnL mentioned it's only used in a single class. It also doesn't do more than add those two fields.